source/classes.tex

```diff
@@ -2264,11 +2264,15 @@
 \tcode{virtual}\iref{class.virtual}
 or pure
 \tcode{virtual}\iref{class.abstract}.
+\begin{note}
 If the destructor of a class is virtual and
 any objects of that class or any derived class are created in the program,
-the destructor shall be defined.
-If a class has a base class with a virtual destructor, its  destructor
-(whether user- or implicitly-declared) is virtual.
+the destructor must be defined; see~\ref{basic.def.odr}.
+\end{note}
+\begin{note}
+If a class \tcode{C} has a base class with a virtual destructor,
+the destructor of \tcode{C} (whether user- or implicitly-declared) is virtual\iref{class.virtual}.
+\end{note}
```

Comment:
This seems entirely wrong, both before and after these changes. If the destructor is virtual and not pure virtual, it must be defined regardless of whether any objects exist. I think the intent is to talk about *pure* virtual destructors here, in which case "any objects of that class or any derived class are created" is not the right constraint either! Merely deriving from the class with a class that has a non-pure virtual destructor is enough to (indirectly) require the pure virtual base class destructor to exist, because the derived class destructor is virtual and therefore odr-used, and it in turn odr-uses the base class destructor.

So I think instead we should say something like:

> [*Note*: A pure virtual destructor must still be defined in most cases, because it is invoked by the destructor of any derived class.]